.github/workflows/workflow.yaml

```diff
@@ -43,16 +43,16 @@ jobs:
     steps:
     - uses: actions/checkout@v4
       with:
-        fetch-depth: 0 #required to be able to generate prerelease tags
+        fetch-depth: 0 # Required to be able to generate prerelease tags
     - uses: actions/setup-go@v5
       with:
         go-version-file: go.mod
-    - uses: google-github-actions/auth@v2
+    - name: Build
+      run: make debug-workflow all
+    - uses: google-github-actions/auth@v2 # This BREAKS the git checkout/tags/history, so must run after builds!
```

Comment:
This drove me insane trying to find the issue here. Still don't know HOW it breaks things, but good enough for now.